tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -51,6 +51,34 @@ async def test_string_handle_takes_bytes(dut):
     assert val == b"bytes"
 
 
+@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith(("icarus", "ghdl")) or
+             cocotb.LANGUAGE in ["verilog"] and cocotb.SIM_NAME.lower().startswith(("riviera")))
+async def test_string_ansi_color(dut):
+    """Check how different simulators treat ANSI-colored strings, see gh-2328"""
+    teststr = "\x1b[33myellow\x1b[49m\x1b[39m"
+    asciival_sum = 0
+    for char in teststr:
+        asciival_sum += ord(char)
+    await cocotb.triggers.Timer(10, 'ns')
+    dut.stream_in_string.value = bytes(teststr.encode("ascii"))
+    await cocotb.triggers.Timer(10, 'ns')
+    val = dut.stream_in_string.value
+    assert isinstance(val, bytes)
+    if cocotb.LANGUAGE in ["vhdl"] and cocotb.SIM_NAME.lower().startswith("riviera"):
+        # Riviera-PRO doesn't return anything with VHDL:
+        assert val == b""
+        # ...and the value shows up differently in the HDL:
+        assert dut.stream_in_string_asciival_sum.value == 1587
```

Comment:
Perhaps to make the cause clearer:
```suggestion
        assert dut.stream_in_string_asciival_sum.value == sum(ord(char) for char in teststr.replace('\x1b', '\0')
```
Although that `\0` will maybe become a space if you change the `if` I comment on above.

Reply:
Good idea, I also replaced the other assertion.